app/src/main/kotlin/net/primal/android/nostr/ext/DirectMessageEvents.kt

```
@@ -35,18 +35,18 @@ fun PrimalEvent.asMessageConversationsSummary(): ConversationsSummary {
     return ConversationsSummary(summaryPerParticipantId = map)
 }
 
-fun List<NostrEvent>.mapAsMessageDataPO(userId: String, nsec: String) =
+fun List<NostrEvent>.mapAsMessageDataPO(userId: String, nsec: String?) =
     mapNotNull { it.mapAsMessageDataPO(userId = userId, nsec = nsec) }
 
-fun NostrEvent.mapAsMessageDataPO(userId: String, nsec: String): DirectMessageData? {
+fun NostrEvent.mapAsMessageDataPO(userId: String, nsec: String?): DirectMessageData? {
     val senderId = this.pubKey
     val receiverId = this.tags.findFirstProfileId() ?: return null
     val participantId = if (senderId != userId) senderId else receiverId
 
     val decryptedMessage = runCatching {
         CryptoUtils.decrypt(
             message = this.content,
-            privateKey = nsec.bechToBytesOrThrow(hrp = "nsec"),
+            privateKey = nsec?.bechToBytesOrThrow(hrp = "nsec")!!, /* TODO(marko): what to do here? */
```

I would throw the missing key exception if it's `null`. We have `getOrElse` in place so we are good. With this approach we can later check the exception in the block below and customize the content if we want.